runtime/context.go

```diff
@@ -35,3 +36,21 @@ func AnnotateContext(ctx context.Context, req *http.Request) context.Context {
 	}
 	return metadata.NewContext(ctx, metadata.Pairs(pairs...))
 }
+
+type ServerMetadata struct {
+	HeaderMD  metadata.MD
+	TrailerMD metadata.MD
+}
+
+type serverMetadataKey struct{}
+
+// NewServerMetadataContext creates a new context with ServerMetadata
+func NewServerMetadataContext(ctx context.Context, md *ServerMetadata) context.Context {
```

Comment:
IIUC, you no longer need to pass a `*ServerMetadata`.  You can pass just `ServerMetadata`.